dash_auth/oidc_auth.py

```diff
@@ -253,29 +257,35 @@ def logout(self):  # pylint: disable=C0116
         return page
 
     def callback(self, idp: str):  # pylint: disable=C0116
-        """Do the OIDC dance."""
+        """Handle the OIDC dance and post-login actions."""
         if idp not in self.oauth._registry:
             return f"'{idp}' is not a valid registered idp", 400
 
         oauth_client = self.get_oauth_client(idp)
         oauth_kwargs = self.get_oauth_kwargs(idp)
         try:
-            token = oauth_client.authorize_access_token(
-                **oauth_kwargs.get("authorize_token_kwargs", {}),
+            self.token = oauth_client.authorize_access_token(
```

Comment:
We shouldn't set the token on the instance as it is user specific but the instance may be used for several users. Just pass the token to the next function if required.